subgraphs/aave-v2/src/modules/Revenue.ts

```diff
@@ -0,0 +1,178 @@
+  const financialDailySnapshot = getOrCreateFinancialsDailySnapshot(event.block);
+
+  if (!variableTokenSupply.reverted && !stableTokenSupply.reverted) {
+    log.info(
+      "IN REPAY FOR MARKET " +
+        market.id +
+        " STABLE OR VARIABLE TOKEN SUPPLIES " +
+        stableTokenSupply.value.toString() +
+        " " +
+        market.totalStableValueLocked.toString() +
+        " " +
+        variableTokenSupply.value.toString() +
+        " " +
+        market.totalVariableValueLocked.toString(),
+      []
+    );
+    market.totalVariableValueLocked = variableTokenSupply.value;
+    market.totalStableValueLocked = stableTokenSupply.value;
+  } else {
+    log.info(
+      "IN REPAY FOR MARKET " +
+        market.id +
+        " COULD NOT GET STABLE OR VARIABLE TOKEN SUPPLIES " +
+        stableTokenSupply.reverted.toString() +
+        " " +
+        variableTokenSupply.reverted.toString(),
+      []
+    );
+  }
+  // Subtract prior market total fees protocol.totalRevenueUSD
+  const protocolId = getProtocolIdFromCtx();
+  const protocol = getOrCreateLendingProtocol(protocolId);
+  log.info(
+    "SUBTRACTING MARKET FROM PROTOCOL TOTAL FEES " +
+      protocol.cumulativeTotalRevenueUSD.toString() +
+      " - " +
+      market.cumulativeTotalRevenueUSD.toString(),
+    []
+  );
+
+  // Get the protocol revenues/fees subtracting the market values before calculation
+  const protoMinusMarketProtoRevenue = protocol.cumulativeProtocolSideRevenueUSD.minus(
+    market.cumulativeProtocolSideRevenueUSD
+  );
+  const protoMinusMarketSupplyRevenue = protocol.cumulativeSupplySideRevenueUSD.minus(
+    market.cumulativeSupplySideRevenueUSD
+  );
+  const protoMinusMarketFees = protocol.cumulativeTotalRevenueUSD.minus(
+    market.cumulativeTotalRevenueUSD
+  );
+
+  const rewardTokens = market.rewardTokens;
+  if (!rewardTokens) return;
+
+  const stableBorrowRate = InterestRate.load(rewardTokens[0])!;
+  const variableBorrowRate = InterestRate.load(rewardTokens[1])!;
+  if (!stableBorrowRate || !variableBorrowRate) return;
+
+  // Multiply total Variable value Locked in USD by market.variableBorrowRate
+  const varAmountUSD = market.inputTokenPriceUSD.times(
+    market.totalVariableValueLocked
+      .toBigDecimal()
+      .div(constants.BIGINT_TEN.pow(token.decimals as u8).toBigDecimal())
+  );
+
+  const varFees = varAmountUSD.times(variableBorrowRate.rate);
+
+  // Multiply total Stable value Locked in USD by market.variableStableRate
+  const staAmountUSD = market.inputTokenPriceUSD.times(
+    market.totalStableValueLocked
+      .toBigDecimal()
+      .div(constants.BIGINT_TEN.pow(token.decimals as u8).toBigDecimal())
+  );
+  const staFees = staAmountUSD.times(stableBorrowRate.rate);
+
+  // Add these values together, save to market and add protocol total
+  let oldCumulativeTotalRevenueUSD = market.cumulativeTotalRevenueUSD;
+  market.cumulativeTotalRevenueUSD = staFees.plus(varFees).truncate(3);

```

Comment:
This is incorrect, you cannot calculate total fee amounts by multiplying the current interest rate with current balance. For variable debt you could get this from the variable debt index, but for stable debt there is no way to get this data as far as I know.

Reply:
This also breaks the rest of the revenue (snapshot) fields because they're based on the assumption that this field is cumulative.

Reply:
I'm gong to resolve these for now but @dmelotik feel free to refer back to these comments when you work on fixing these issues